gradio/flagging.py

```diff
@@ -344,6 +346,50 @@ def flag(
         return line_count
 
 
+class ChatCSVLogger(FlaggingCallback):
+    """
+    Flagging callback for chat conversations.
+    Flagged conversations and like/dislike reactions are logged to a CSV file on the machine running the gradio app.
+    """
+
+    def __init__(self):
+        pass
+
+    def setup(self, flagging_dir: str | Path):
+        self.flagging_dir = flagging_dir
+        os.makedirs(flagging_dir, exist_ok=True)
+
+    def flag(
+        self,
+        like_data: LikeData,
+        messages: list,
+    ) -> int:
+        flagging_dir = self.flagging_dir
+        log_filepath = Path(flagging_dir) / "log.csv"
+        is_new = not Path(log_filepath).exists()
+
+        feedback = (
+            "Like"
+            if like_data.liked == True  # noqa: E712
+            else "Dislike"
+            if like_data.liked == False  # noqa: E712
```

Comment:
why do you have the noqas? Shouldn't it be:
```suggestion
            if like_data.liked is True  # noqa: E712
            else "Dislike"
            if like_data.liked is False  # noqa: E712
```